### PR TITLE
[tests/nnfw_api] Add Conv2D test for train

### PR DESCRIPTION
**Pull request by YongseopKim**
This commit adds a Conv2D test for train as an example.

ONE-DCO-1.0-Signed-off-by: Yongseop Kim <yons.kim@samsung.com>

---

draft: https://github.com/Samsung/ONE/pull/12665

### CODE REVIEW

**Comment by YongseopKim** [comment]
PTAL @Samsung/one_onert

**Review by jyoungyun — APPROVED** [review]
LGTM

**Comment by jyoungyun** [comment]
(suggestion)

How about checking the BackProp Tensor result as well as the Loss value?

**Comment by YongseopKim** [comment]
@jyoungyun can we access weight values or back prop values? AFAIK, apis related to it are not there.

**Comment by jyoungyun** [comment]
@YongseopKim 

> can we access weight values or back prop values? AFAIK, apis related to it are not there.

Oh, it's my mistake. I wanted to tell you about the updated Weight and Bias tensors. After the training process, we can compare the updated Weight and Bias value with the expected value. This will help check whether the back-propagation works well or not.

**Comment by YongseopKim** [comment]
And we get an idea to train by two epochs to apply the first back-progation into loss value.

**Comment by YongseopKim** [comment]
anyway, PTAL @Samsung/one_onert

**Comment by jyoungyun** [comment]
It is possible to check whether the back-propagation works well through more than one epoch even if it does not compare the weight and bias tensors with the expected value.

File: tests/nnfw_api/src/GenModelTests/one_op_trains/Conv2D.test.cc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * limitations under the License.
+ */
+
+#include "GenModelTrain.h"
+
+TEST_F(GenModelTrain, OneOp_Conv2D)
+{
+  CirclePlusGen cgen;
+
+  int in = cgen.addTensor({{1, 5, 5, 1}, circle::TensorType::TensorType_FLOAT32});
```
Comment (zetwhite) [inline tests/nnfw_api/src/GenModelTests/one_op_trains/Conv2D.test.cc:23]: Is there a reason to use `int` here? 
Since `addTensor` return uint32_t, I thought that the type might be `uint32_t` either.

https://github.com/Samsung/ONE/blob/31c728d79bd8e3c91fb0b85a2cb7251c22fc9f82/tests/nnfw_api/lib/CircleGen.cc#L40C1-L41C1 

```suggestion
  uint32_t in = cgen.addTensor({{1, 5, 5, 1}, circle::TensorType::TensorType_FLOAT32});
```

**Review by zetwhite — APPROVED** [review]
LGTM :+1:
